[docs] EKS & AKS auto-discovery docs (#17628)
gravitational/teleport: docs/pages/kubernetes-access/discovery/azure.mdx

        "Microsoft.ContainerService/managedClusters/pods/read",
        "Microsoft.ContainerService/managedClusters/authorization.k8s.io/selfsubjectaccessreviews/write",
        "Microsoft.ContainerService/managedClusters/authorization.k8s.io/selfsubjectrulesreviews/write",
      ],
      "NotDataActions": [],
      "assignableScopes": [
          "/subscriptions/{subscription_id}"
      ]
  }
```

Replace the `{subscription_id}` with the desired Subscription ID or a wildcard if 
you want to guarantee permissions on all subscriptions.

  </TabItem>

  <TabItem label="Azure AD + Kubernetes RBAC">

When using Azure AD authentication with Kubernetes RBAC mode, Azure is responsible 
for user authentication using AD credentials, but permissions management is Kubernetes' 
RBAC responsibility.

Therefore, for Teleport to work correctly, you must create the Kubernetes `ClusterRole` 
and `ClusterRoleBinding` resources on each discovered cluster. 
The `ClusterRoleBinding` must bind the `ClusterRole` to one of the AD groups 
configured in the Teleport identity.

Teleport can automatically create the `ClusterRole` and `ClusterRoleBinding` resources 
in the following cases:

> - Teleport identity has access to the static [cluster administrator credentials](https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/list-cluster-admin-credentials) (Local Accounts).

Author: The AD identity Teleport is using. It can be an APP, a user, or a role loaded into a VM..

- Teleport identity belongs to the cluster's administrator group.
- Teleport identity has permissions to create `ClusterRole` and `ClusterRoleBinding` on the 

> cluster and permissions to execute [remote commands](https://learn.microsoft.com/pt-pt/rest/api/aks/managed-clusters/run-command).

Reviewer: ```suggestion
cluster and permissions to execute [remote commands](https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/run-command).
```

On second thought, I'm not sure there's a succinct way to describe this API path within this list, so a link should be okay here. We should still us the `en-us` locale to be consistent with the rest of the docs, though

command against your Teleport Auth Service and save it in `/tmp/token` on the 
machine that will run the Kubernetes Auto-Discovery:

```code
$ tctl tokens add --type=discovery,kube
```

Enabling AKS Auto-Discovery requires that the `discovery_service.azure` section 
include at least one entry and that `discovery_service.azure.types` include `aks`. 
It also requires configuring the `kubernetes_service.resources.tags` to use the same 
labels configured at `discovery_service.azure.tags` or a subset of them to make 
the Kubernetes Service listen to the dynamic resources created by the Discovery 
Service.

```yaml
version: v2
teleport:
  join_params:
    token_name: "/tmp/token"
    method: token
  auth_servers:
  - "teleport.example.com:3080"
auth_service:
  enabled: off
proxy_service:
  enabled: off
ssh_service:
  enabled: off
discovery_service:
  enabled: "yes"
  azure:
  - types: ["aks"]
    regions: ["*"]
    subscriptions: ["*"]
    resource_groups: ["*"]
    tags:
      "env": "prod"
kubernetes_service:
  enabled: "yes"
  resources:
  - labels:
      "env": "prod" # Match Kubernetes Cluster labels specified earlier
```

> Once configured, Teleport can be started and AKS clusters matching the tags and regions

Reviewer: ```suggestion
Once you have added this configuration, start Teleport. AKS clusters matching the tags and regions
```

---
title: Teleport AKS Auto-Discovery (Preview)
description: Auto-Discovery of AKS clusters in Azure cloud.
---

> Similarly to [EKS Auto-Discovery](./aws.mdx), AKS Auto-Discovery can automatically 

Reviewer 2: ```suggestion
AKS Auto-Discovery can automatically 
```
